### PR TITLE
Consolidate README guides into one file

### PR DESCRIPTION
**Pull request by MessagingMoore**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,15 +3,84 @@
 This repository contains the REST API running as a PubNub function that is required for
 the [ChatEngine Framework](https://github.com/pubnub/chat-engine/) to operate.
 
-## Setup
+## ChatEngine Application and Key Provisioning Guide
 
-First, configure your application in the PubNub administration UI.
+To setup ChatEngine on PubNub, one must first setup a new PubNub App. The following outlines how to manually setup a PubNub ChatEngine Key.
```
Comment (polarweasel) [inline README.md]: "To set up ... you must first..."

File: README.md [hunk 2]
```diff
@@ -3,15 +3,84 @@
...
 
-[How to Setup your account for ChatEngine](guides/ChatEngine_Application_and_Key_Provisioning.md)
+Setup an account, create a set up pubsub keys, configure these keys to
```
Comment (polarweasel) [inline README.md]: Incomplete sentence. Also, you "set up" an account, but you deliver the "setup" for a joke.

Comment (polarweasel) [reply README.md]: "pub/sub keys"

File: README.md [hunk 3]
```diff
@@ -3,15 +3,84 @@
...
+## ChatEngine Application and Key Provisioning Guide
 
-First, configure your application in the PubNub administration UI.
+To setup ChatEngine on PubNub, one must first setup a new PubNub App. The following outlines how to manually setup a PubNub ChatEngine Key.
 
-[How to Setup your account for ChatEngine](guides/ChatEngine_Application_and_Key_Provisioning.md)
+Setup an account, create a set up pubsub keys, configure these keys to 
 
-Then, build the function in this repository and upload it.
+### Basic Steps
```
Comment (polarweasel) [inline README.md]: Creating a new app, maybe? Or at least, basic steps to what?

File: README.md [hunk 4]
```diff
@@ -3,15 +3,84 @@
...
+1. Navigate to [PubNub's Portal](pubnub.com), sign in, and select the _Create New App_ button at the top-right of the logged in user's app page.
```
Comment (polarweasel) [inline README.md]: dashboard.pubnub.com

Comment (polarweasel) [reply README.md]: and you can delete "at the top-right of the logged in user's app page"

File: README.md [hunk 5]
```diff
@@ -3,15 +3,84 @@
...
+1. Enter an App name (e.g. `ChatEngine Application`) and click the _Create_ button.
+1. Select the newly made App icon and navigate to the _Demo Keyset_.
+1.  The following setting need to be updated on the key:
```
Comment (polarweasel) [inline README.md]: Update the following setting on the keyset:

File: README.md [hunk 6]
```diff
@@ -3,15 +3,84 @@
...
+|:--------------|:--------:|
+| Enable Wildcard Subscribe | `Enabled` |
+
+| **PubNub Functions** | `Enabled` |
+|:--------------|:--------:|
+
+| **Access Manager** | `Enabled` |
+|:--------------|:--------:|
+
+## ChatEngine Server Provisioning Guide
```
Comment (polarweasel) [inline README.md]: Provisioning ChatEngine Server (and then you can delete the "basic steps" h3, and move everything else up a level)

File: README.md [hunk 7]
```diff
@@ -3,15 +3,84 @@
...
+
+### Basic Steps
+
+#### I. Bundle Server Code
+1. Clone the `chat-engine-server` repo and run `npm install`
```
Comment (polarweasel) [inline README.md]: period at end

File: README.md [hunk 8]
```diff
@@ -3,15 +3,84 @@
...
+1. Created a bundled function at `app/functions/server.js` by running  `npm run build`
```
Comment (polarweasel) [inline README.md]: period at end

reverse the order: Run `npm run build` to create a bundled function at `app/functions/server.js`.

all over the place: replace double spaces with single

File: README.md [hunk 9]
```diff
@@ -3,15 +3,84 @@
...
+1. Run `npm test` to verify that the bundle is correct
```
Comment (polarweasel) [inline README.md]: period at end

File: README.md [hunk 10]
```diff
@@ -3,15 +3,84 @@
...
+1. Copy the code to your clipboard for later use.
+
+#### II. Create a PubNub REST Function 
+
+1. Go to the ChatEngine App and Key setup in the previous guide: [ChatEngine Application and Key Provisioning Guide]()
```
Comment (polarweasel) [inline README.md]: ...and do what?

"On your [PubNub Admin Dashboard](https://dashboard.pubnub.com), go the ChatEngine app key you set up in the [previous guide]()."

File: README.md [hunk 11]
```diff
@@ -3,15 +3,84 @@
...
+1. Select the _Functions_ tab on the top left-hand side of the ChatEngine Application.
```
Comment (polarweasel) [inline README.md]: "Click the Functions tab on the left side of the page."

File: README.md [hunk 12]
```diff
@@ -3,15 +3,84 @@
...
+1. Fill out the following text fields and click the _Create New Module_ button:
```
Comment (polarweasel) [inline README.md]: "...and click Create New Module:"

File: README.md [hunk 13]
```diff
@@ -3,15 +3,84 @@
...
+
+	| Field	| Value |
+	|:------|:-----:|
+	| _Module Name_: | `ChatEngine Server` |
+	| _Module Description_: | `REST Function that powers PubNub ChatEngine` |
+
+	
+1. Within the ChatEngine Module, click the _+ create_ button and fill in the following text fields:
```
Comment (polarweasel) [inline README.md]: "...click Create and..."

File: README.md [hunk 14]
```diff
@@ -3,15 +3,84 @@
...
+
+	| Field	| Value |
+	|:------|:-----:|
+	| _Function Name_ | `ChatEngine Server`|
+	| _Select Event Type_ | `On Request` |
+	| _URI path_ | `chat-engine-server` |
+
+1. Press the _Create_ Function button to navigate to the PubNub Functions Console.
```
Comment (polarweasel) [inline README.md]: "Click Create Function to navigate..."

File: README.md [hunk 15]
```diff
@@ -3,15 +3,84 @@
...
+1. With the console open, copy & paste the server code at `app/functions/server.js` into the console editor.
```
Comment (polarweasel) [inline README.md]: "Copy and paste the server code..., and click Save."

Delete the next line.

File: README.md [hunk 16]
```diff
@@ -3,15 +3,84 @@
...
+	| Field	| Value |
+	|:------|:-----:|
+	| _Function Name_ | `ChatEngine Server`|
+	| _Select Event Type_ | `On Request` |
+	| _URI path_ | `chat-engine-server` |
+
+1. Press the _Create_ Function button to navigate to the PubNub Functions Console.
+1. With the console open, copy & paste the server code at `app/functions/server.js` into the console editor. 
+1. Click the _Save_ button at the top left-hand side of the console editor.
+1. Click the _My Secrets_ button, enter the following, and click the _Save_ button (get the `secretKey` from the App and Key [page](admin.pubnub.com)):
```
Comment (polarweasel) [inline README.md]: "Click My Secrets, and enter the following (get the `secretKey` from the App and Key [page](dashboard.pubnub.com)):

...

1. Click Save, and return to the console."

File: README.md [hunk 17]
```diff
@@ -3,15 +3,84 @@
...
+
+	| Field	| Value |
+	|:------|:-----:|
+	| _Enter Key_ | `secretKey`|
+	| _Enter Value_ | `__COPY_AND_PASTE_PUBNUB_SECRET_KEY__` |
+	
+1. Returning to the console, click the _Start_ Function button in the top right-hand corner of the console.
```
Comment (polarweasel) [inline README.md]: Click Start Function, and you're done!

File: README.md [hunk 18]
```diff
@@ -3,15 +3,84 @@
...
+1. Done!
```
Comment (polarweasel) [inline README.md]: Delete

**Review by polarweasel — CHANGES_REQUESTED** [review]

File: README.md [hunk 19]
```diff
@@ -39,48 +37,50 @@ Setup an account, create a set up pubsub keys, configure these keys to
 | **Access Manager** | `Enabled` |
 |:--------------|:--------:|
 
-## ChatEngine Server Provisioning Guide
+## Provisioning the ChatEngine Framework Function
+
+### I. Bundle the server Server Code.
```
Comment (polarweasel) [inline README.md]: Might want to call it "Part 1: Bundle the..." (and the same for the next one)

File: README.md [hunk 20]
```diff
@@ -39,48 +37,50 @@ Setup an account, create a set up pubsub keys, configure these keys to
...
 
-### Basic Steps
+The REST api is constructed of several files to improve understandability. The files
```
Comment (polarweasel) [inline README.md]: API all-caps

"composed of" or "made up of"

File: README.md [hunk 21]
```diff
@@ -39,48 +37,50 @@ Setup an account, create a set up pubsub keys, configure these keys to
...
-1. Clone the `chat-engine-server` repo and run `npm install`
-1. Created a bundled function at `app/functions/server.js` by running  `npm run build`
-1. Run `npm test` to verify that the bundle is correct
+1. Clone the `chat-engine-server` repo and run `npm install`.
+1. Run `npm run build` to create a bundled function at `app/functions/server.js`.
+1. Run `npm test` to verify that the bundle is correct.
 1. Copy the code to your clipboard for later use.
 
-#### II. Create a PubNub REST Function 
+### II. Create a PubNub REST Function.
```
Comment (polarweasel) [inline README.md]: no terminal punctuation in headings!

File: README.md (hunk 19 above)
Comment (polarweasel) [reply README.md]: Also: no terminal punctuation in headings

File: README.md [hunk 22]
```diff
@@ -39,48 +37,50 @@ Setup an account, create a set up pubsub keys, configure these keys to
...
 
-1. Go to the ChatEngine App and Key setup in the previous guide: [ChatEngine Application and Key Provisioning Guide]()
-1. Select the _Functions_ tab on the top left-hand side of the ChatEngine Application.
-1. Fill out the following text fields and click the _Create New Module_ button:
+1. On your [PubNub Admin Dashboard](dashboard.pubnub.com), go the ChatEngine app key you set up in the previous section to upload your Function.
```
Comment (polarweasel) [inline README.md]: Need a full URL

**Review by polarweasel — CHANGES_REQUESTED** [review]
added inline comments....

**Review by polarweasel — APPROVED** [review]
👍 LGTM

**Review by polarweasel — APPROVED** [review]
LGTM with write access this time

**Review by nishith-pubnub — APPROVED** [review]

**Review by madisonsmith — APPROVED** [review]

**Pull request merged**